src/Analyzers/CSharp/Analyzers/UsePatternCombinators/CSharpUsePatternCombinatorsAnalyzer.cs

```diff
@@ -0,0 +1,169 @@
+                var (v, _) when IsConstant(v) => ConstantResult.Left,
+                _ => ConstantResult.None,
+            };
+        }
+
+        private static AnalyzedPattern? ParseRelationalPattern(IBinaryOperation op)
+        {
+            return DetermineConstant(op) switch
+            {
+                ConstantResult.Left when op.LeftOperand.Syntax is ExpressionSyntax left
+                    // We need to flip the operator if the constant is on the left-hand-side.
+                    // This is because relational patterns only come in the prefix form.
+                    // For instance: `123 > x` would be rewritten as `x is < 123`.
+                    => new Relational(Flip(op.OperatorKind), left, op.RightOperand),
+                ConstantResult.Right when op.RightOperand.Syntax is ExpressionSyntax right
+                    => new Relational(op.OperatorKind, right, op.LeftOperand),
+                _ => null
+            };
+        }
+
+        private static AnalyzedPattern? ParseConstantPattern(IBinaryOperation op)
+        {
+            return DetermineConstant(op) switch
+            {
+                ConstantResult.Left when op.LeftOperand.Syntax is ExpressionSyntax left
+                    => new Constant(left, op.RightOperand),
+                ConstantResult.Right when op.RightOperand.Syntax is ExpressionSyntax right
+                    => new Constant(right, op.LeftOperand),
+                _ => null
+            };
+        }
+
+        private static bool IsRelationalOperator(BinaryOperatorKind operatorKind)
+        {
+            switch (operatorKind)
+            {
+                case LessThan:
+                case LessThanOrEqual:
+                case GreaterThanOrEqual:
+                case GreaterThan:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        /// <summary>
+        /// Changes the direction the operator is pointing at.
+        /// </summary>
+        /// <remarks>
+        /// Relational patterns only come in the prefix form so we'll have to
+        /// flip the operator if the constant happens to be on the left-hand-side.
+        /// </remarks>
+        public static BinaryOperatorKind Flip(BinaryOperatorKind operatorKind)
+        {
+            return operatorKind switch
+            {
+                LessThan => GreaterThan,
+                LessThanOrEqual => GreaterThanOrEqual,
+                GreaterThanOrEqual => LessThanOrEqual,
+                GreaterThan => LessThan,
+                var v => throw ExceptionUtilities.UnexpectedValue(v)
+            };
+        }
+
+        private static bool IsConstant(IOperation operation)
+        {
+            // By-design, constants will not propagate to conversions.
+            return operation is IConversionOperation op
+                ? IsConstant(op.Operand)
+                : operation.ConstantValue.HasValue;
```

Comment:
i don't understand this logic.  unwrapping the conversion seems weird.  or maybe i'm just not understand the comment.  can you explain why this works witha few examples?

Reply:
this is by design. see https://github.com/dotnet/roslyn/issues/38272

Reply:
ok, do you have good test coverage here (for both implicit and explicit conversions)?